src/app/hooks/useApolloQuery.tsx

```diff
@@ -16,13 +16,24 @@
  * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
+import { apiName } from '@/lib/gql';
+import {
+	DocumentNode,
+	OperationVariables,
+	QueryHookOptions,
+	QueryResult,
+	TypedDocumentNode,
+	useQuery,
+} from '@apollo/client';
 
-import { gql } from '@/__generated__';
-
-const CLINICAL_SCHEMA_VERSION_QUERY = gql(`
-	query ClinicalSchemaVersion {
-		clinicalSubmissionSchemaVersion
-	}
-`);
-
-export default CLINICAL_SCHEMA_VERSION_QUERY;
+export const useApolloQuery = <TData, TVariables>(
+	query: {
+		gql: DocumentNode | TypedDocumentNode<TData, TVariables>;
+		api: keyof typeof apiName;
+	},
+	options?: QueryHookOptions<any, OperationVariables> | undefined,
+): QueryResult<TData, TVariables> => {
+	const mergedOptions = { context: { apiName: query.api }, ...options };
+	// @ts-expect-error
```

Comment:
@joneubank if you fancy ;) 
TS doesn't like that I'm spreading options into a new object. There's a `NoInfer` types from apollo in use, I can't understand it.